core/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -222,6 +298,7 @@ public void removeBan(TaskId parentTaskId) {
 
     @Override
     public void clusterChanged(ClusterChangedEvent event) {
+        lastDiscoveryNodes = event.state().getNodes();
```

Comment:
Answering my own question - no - we didn't need it before. I should read more closely next time.